### PR TITLE
Configurable Chase Camera

### PR DESCRIPTION
**Pull request by flyingk**
This PR extends the previous runway camera contribution by adding an option to position the chase camera relative to the aircraft such that it moves with it and points at the specified direction in local aircraft axes.

This allows to quickly switch pre-programmed camera angles of the flying aircraft.

A new file was added to group the camera callbacks.

### CODE REVIEW

**Comment by jason-watkins** [comment]
It looks like you're trying to include the C client header in one of the plugin source files... Was that intentional for some reason?

Looks good otherwise. I'll do a detailed review once all of the builds are working.

**Comment by flyingk** [comment]
I am using the enum VIEW_TYPE, so to ensure consistency I included the client header. How would you propose to do it to ensure compatibility? Obviously, the enum can be copied into the plugin header, but then changes have to be applied in both locations.

**Comment by jason-watkins** [comment]
Ah, I see. I'd prefer not to do that for two reasons. One, I just don't like the cross dependency from a build perspective. Second, I like the plugin code to stay C++, and I hate C-style enums in C++ code.

I don't think there's any real problem with copying the enum over to the plugin header (and making it an `enum class` in the process), especially since we at least nominally have to keep all of the clients up to date so reusing the C client enum doesn't actually solve the issue of consistency completely.

**Comment by flyingk** [comment]
Here is the msg definition for the extended camera for the wiki:
[xplane_camera_msg.xlsx](https://github.com/nasa/XPlaneConnect/files/3845594/xplane_camera_msg.xlsx)

File: xpcPlugin/CameraCallbacks.cpp [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+			struct CameraProperties* campos = (struct CameraProperties*)inRefcon;
+			
+			// camera position
+			double clat = campos->loc[0];
+			double clon = campos->loc[1];
+			double calt = campos->loc[2];
+			
+            double cX;
+            double cY;
+            double cZ;
```
Comment (jason-watkins) [inline xpcPlugin/CameraCallbacks.cpp]: Mismatched indentation here. Probably mixed tabs and spaces?

File: xpcPlugin/CameraCallbacks.cpp [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+			
+			if(campos->direction[1] < -180)
+				outCameraPosition->pitch	= 0;
+			else
+				outCameraPosition->pitch	= the + campos->direction[1];
+			
+			if(campos->direction[2] < -180)
+				outCameraPosition->heading	= 0;
+			else
+				outCameraPosition->heading	= psi + campos->direction[2];
```
Comment (jason-watkins) [inline xpcPlugin/CameraCallbacks.cpp]: Braces on all if/else blocks please

File: xpcPlugin/MessageHandlers.h [hunk 3]
```diff
@@ -61,7 +78,8 @@ namespace XPC
 		static void HandleXPlaneData(const Message& msg);
 		static void HandleUnknown(const Message& msg);
 		
-		static int CamFunc( XPLMCameraPosition_t * outCameraPosition, int inIsLosingControl, void *inRefcon);
+        static int CamCallback_RunwayCam( XPLMCameraPosition_t * outCameraPosition, int inIsLosingControl, void *inRefcon);
+        static int CamCallback_ChaseCam( XPLMCameraPosition_t * outCameraPosition, int inIsLosingControl, void *inRefcon);
```
Comment (jason-watkins) [inline xpcPlugin/MessageHandlers.h]: More mismatched indentation

**Comment by flyingk** [comment]
this all works fine on Mac, so #75 can be closed as well.

**Comment by jason-watkins** [comment]
Looks good. Thanks as always for the contribution!

**Pull request merged**